server/py/framework/db/sqldb/db.py

```diff
@@ -3216,52 +3215,54 @@ def _calculate_alert_activations_counters(
         project_to_job_alerts_count = collections.defaultdict(int)
         project_to_other_alerts_count = collections.defaultdict(int)
 
-        project_filter_conditions = self._generate_project_filter_conditions(
-            projects_with_creation_time
-        )
-
         last_day = mlrun.utils.datetime_now() - timedelta(hours=24)
 
-        query_results = (
-            session.query(
-                AlertActivation.project,
-                func.count(
-                    case(
-                        (
-                            AlertActivation.entity_kind
-                            == mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
-                            1,
-                        )
+        # construct a base query to count different types of alert activations
+        query = session.query(
+            AlertActivation.project,
+            func.count(
+                case(
+                    (
+                        AlertActivation.entity_kind
+                        == mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
+                        1,
                     )
-                ).label("model_endpoint_alerts_count"),
-                func.count(
-                    case(
-                        (
-                            AlertActivation.entity_kind
-                            == mlrun.common.schemas.alert.EventEntityKind.JOB,
-                            1,
-                        )
+                )
+            ).label("model_endpoint_alerts_count"),
+            func.count(
+                case(
+                    (
+                        AlertActivation.entity_kind
+                        == mlrun.common.schemas.alert.EventEntityKind.JOB,
+                        1,
                     )
-                ).label("job_alerts_count"),
-                func.count(
-                    case(
-                        (
-                            ~AlertActivation.entity_kind.in_(
-                                [
-                                    mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
-                                    mlrun.common.schemas.alert.EventEntityKind.JOB,
-                                ]
-                            ),
-                            1,
-                        )
+                )
+            ).label("job_alerts_count"),
+            func.count(
+                case(
+                    (
+                        AlertActivation.entity_kind.not_in(
+                            [
+                                mlrun.common.schemas.alert.EventEntityKind.MODEL_ENDPOINT_RESULT,
+                                mlrun.common.schemas.alert.EventEntityKind.JOB,
+                            ]
+                        ),
+                        1,
                     )
-                ).label("other_alerts_count"),
+                )
+            ).label("other_alerts_count"),
```

Comment:
Do we really need labels here?

Reply:
its just to improve readability, which can be helpful for debugging or any future enhancements where these labels might be referenced.

Reply:
@Yacouby maybe worth leaving comment then so next developer don't remove it